src/core/openthread.cpp

```diff
@@ -66,9 +66,6 @@ static otDEFINE_ALIGNED_VAR(sInstanceRaw, sizeof(otInstance), uint64_t);
 otInstance *sInstance = NULL;
 #endif
 
-void OT_CDECL operator delete(void *, size_t) throw() { }
-void OT_CDECL operator delete(void *) throw() { }
```

Comment:
Well, the scope of your delete implementations is not corresponding with the existing new implementation. Our code should either take the OT's new/delete or none of them. Currently, we can control which new should be used though new.hpp, but we cannot do that for delete's implementation since it's dumped into openthread.cpp. This is bad coding and the excuse that Windows needs it is not valid; a platform/user should be able to either use OT's new/delete or not. 
Your code delete implementation location is not acceptable.